sdk/python/feast/loaders/abstract_producer.py

```diff
@@ -129,7 +129,12 @@ def flush(self, timeout: Optional[int]):
         Returns:
             int: Number of messages still in queue.
         """
-        return self.producer.flush(timeout=timeout)
+        messages = self.producer.flush(timeout=timeout)
```

Comment:
Might be better to fail fast using produce() callbacks rather than wait for the timeout/queue to clear.